addOns/spider/src/main/java/org/zaproxy/addon/spider/parser/SvgHrefParser.java

```diff
@@ -149,7 +205,12 @@ private String extractUrl(Node node) {
 
     @Override
     public boolean canParseResource(HttpMessage message, String path, boolean wasAlreadyConsumed) {
-        return isSvg(message);
+        return isSvg(message)
+                || containsSvg(message, new Source(message.getResponseBody().toString()));
+    }
+
+    private boolean containsSvg(HttpMessage message, Source source) {
```

Comment:
- `static` (`extractUrl` can be also).
- Use a supplier for the `Source` to avoid the parsing if it's not HTML.